src/vos/tests/evt_ctl.c

```diff
@@ -750,6 +751,338 @@ global_teardown(void **state)
 #define NUM_PARTIAL 11
 #define NUM_EXTENTS 30
 
+/* create_expected_data:
+* Input values:  iter_flag, evt_data
+* Input/Output : expval, rev_expval
+*/
+static void
+create_expected_data(int iter_flag, int **evt_data,
+					int *expval, int *rev_expval)
+{
+	int epoch;
+	int expected_size;
+	int count;
+	int loop_count;
+	int offset;
+
+	count = 0;
+	switch (iter_flag) {
+	case EVT_ITER_EMBEDDED:
+		print_message("EVT_ITER_EMBEDDED\n");
+	break;
+	case EVT_ITER_VISIBLE:
+		print_message("EVT_ITER_VISIBLE\n");
+		for (epoch = 1; epoch <= NUM_EPOCHS; epoch++) {
+		if (epoch < NUM_EPOCHS) {
+			expval[epoch] = evt_data[epoch][epoch];
+			count++;
+		} else {
+				for (offset = epoch;
+				offset < NUM_EXTENTS + epoch; offset++) {
+					expval[offset] =
+					evt_data[epoch][offset];
+					count++;
+				}
+			}
+		}
+	break;
+	case EVT_ITER_COVERED:
+		print_message("EVT_ITER_COVERED\n");
+		count = 1;
+		for (epoch = 1; epoch <= NUM_EPOCHS; epoch++) {
+			if (epoch < NUM_EPOCHS) {
+				for (offset = epoch; offset >= 1;
+					offset--) {
+					if (evt_data[offset][epoch+1] != 0) {
+						expval[count] =
+						evt_data[offset][epoch+1];
+						count++;
+					}
+				}
+			} else {
+				for (loop_count = epoch+1;
+				loop_count < NUM_EXTENTS+epoch;
+				loop_count++) {
+				for (offset = epoch-1; offset >= 1;
+					offset--) {
+					if (evt_data[offset][loop_count] != 0) {
+						expval[count] =
+						evt_data[offset][loop_count];
+						count++;
+						}
+					}
+				}
+			}
+		 }
+	break;
+	case (EVT_ITER_VISIBLE | EVT_ITER_COVERED):
+		print_message("EVT_ITER_VISIBLE (COVERED)\n");
+		count = 1;
+		for (epoch = 1; epoch <= NUM_EPOCHS; epoch++) {
+			if (epoch < NUM_EPOCHS) {
+				for (offset = epoch; offset >= 1; offset--) {
+					if (evt_data[offset][epoch] != 0) {
+						expval[count] =
+						evt_data[offset][epoch];
+						count++;
+					}
+				}
+			} else {
+			for (loop_count = epoch;
+				loop_count < NUM_EXTENTS+epoch;
+				loop_count++) {
+				for (offset = epoch;
+				offset >= 1 ; offset--) {
+					if (evt_data[offset][loop_count] != 0) {
+						expval[count] =
+						evt_data[offset][loop_count];
+						count++;
+						}
+					}
+				}
+			}
+		 }
+	break;
+	case (EVT_ITER_SKIP_HOLES|EVT_ITER_VISIBLE):
+		print_message("EVT_ITER_SKIP_HOLES (VISIBLE)\n");
+		for (epoch = 1; epoch <= NUM_EPOCHS; epoch++) {
+			if (epoch < NUM_EPOCHS) {
+				count++;
```

Comment:
would be clearer to move the ++ into the if block here.   Then you don't need the else block.

Reply:
Done.